native/src/main/java/io/ballerina/stdlib/rabbitmq/util/ChannelUtils.java

```diff
@@ -97,20 +98,36 @@ public static Object queueDeclare(Environment environment, BObject clientObj,
                 exclusive = config.getBooleanValue(RabbitMQConstants.QUEUE_EXCLUSIVE);
                 autoDelete = config.getBooleanValue(RabbitMQConstants.QUEUE_AUTO_DELETE);
                 if (config.getMapValue(RabbitMQConstants.QUEUE_ARGUMENTS) != null) {
-                    argumentsMap = (HashMap<String, Object>) config.getMapValue(RabbitMQConstants.QUEUE_ARGUMENTS);
+                    @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+                    HashMap<BString, Object> queueArgs =
+                            (HashMap<BString, Object>) config.getMapValue(RabbitMQConstants.QUEUE_ARGUMENTS);
+                    queueArgs.forEach((k, v) -> argumentsMap.put(k.getValue(), getConvertedValue(v)));
                 }
             }
             channel.queueDeclare(queueName.getValue(), durable, exclusive, autoDelete, argumentsMap);
             RabbitMQMetricsUtil.reportNewQueue(channel, queueName.getValue());
             RabbitMQTracingUtil.traceQueueResourceInvocation(channel, queueName.getValue(), environment);
-        } catch (IOException | ShutdownSignalException exception) {
+        } catch (IOException | ShutdownSignalException | BError exception) {
```

Comment:
If the exception is `BError`, the better way to handle this is to create a new error with adding the caught `BError` as the cause. 

Since this is to catch the `BError` thrown in the `getConvertedValue` method, let's use a separate catch block and just return the `BError` instance.